ui/lib/apps/TopSql/pages/TopSql.tsx

```diff
@@ -0,0 +1,225 @@
+    DEFAULT_TIME_RANGE
+  )
+  const [timeRange, setTimeRange] = useState<TimeRange>(recentTimeRange)
+  const [refreshTimestamp, setRefreshTimestamp] = useState(0)
+  const { seriesData, queryTimestampRange, isLoading } = useSeriesData(
+    instanceId,
+    timeRange,
+    autoRefreshSeconds,
+    refreshTimestamp
+  )
+
+  const setAbsoluteTimeRange = useCallback((tr: [number, number]) => {
+    setAutoRefreshSeconds(0)
+    setTimeRange({ type: 'absolute', value: tr })
+  }, [])
+
+  const handleSetTimeRange = useCallback((v: TimeRange) => {
+    setTimeRange(v)
+    if (v.type === 'recent') {
+      setRecentTimeRange(v)
+    }
+  }, [])
+
+  const handleBrushEnd: BrushEndListener = useCallback((v: XYBrushArea) => {
+    if (v.x) {
+      setAbsoluteTimeRange(
+        v.x.map((d) => Math.ceil(d / 1000)) as [number, number]
+      )
+    }
+  }, [])
+
+  const handleZoomOut = useCallback(() => {
+    const [start, end] = calcTimeRange(timeRange)
+    setAbsoluteTimeRange([start - (end - start) * zoomOutRate, end])
+  }, [timeRange])
+
+  const refreshTimestampRange = useCallback(() => {
+    setRefreshTimestamp(Date.now())
+  }, [])
+
+  const { remainingRefreshSeconds } = useAutoRefreshRemainingSeconds(
+    autoRefreshSeconds,
+    [queryTimestampRange]
+  )
+
+  const handleAutoRefreshSecondsChange = useCallback((v: number) => {
+    setAutoRefreshSeconds(v)
+    setTimeRange(recentTimeRange)
+  }, [])
+
+  return (
+    <div className={styles.container}>
+      <Card>
+        <Space size="middle">
+          <InstanceSelect value={instanceId} onChange={setInstanceId} />
+          <Button.Group>
+            <TimeRangeSelector
+              value={timeRange}
+              onChange={handleSetTimeRange}
+            />
+            <Button icon={<ZoomOutOutlined />} onClick={handleZoomOut} />
+          </Button.Group>
+          <AutoRefreshButton
+            autoRefreshSeconds={autoRefreshSeconds}
+            onAutoRefreshSecondsChange={handleAutoRefreshSecondsChange}
+            remainingRefreshSeconds={remainingRefreshSeconds}
+            isLoading={isLoading}
+            onRefresh={refreshTimestampRange}
+            options={autoRefreshOptions}
+          />
+        </Space>
+      </Card>
+      <Spin spinning={isLoading}>
+        {!isLoading && !seriesData?.length && (
+          <p style={{ marginTop: '100px', textAlign: 'center' }}>
+            {t('top_sql.no_data')}
+          </p>
+        )}
+        <div className={styles.chart_container}>
+          <TopSqlChart
+            onBrushEnd={handleBrushEnd}
+            seriesData={seriesData}
+            timeRange={timeRange}
+            timestampRange={queryTimestampRange}
+          />
+        </div>
+        {!!seriesData?.length && <TopSqlTable data={seriesData} />}
+      </Spin>
+    </div>
+  )
+}
+
+const queryTopSQLDigests = asyncDebounce(
+  (
+    instanceId: InstanceId,
+    windowSize: number,
+    timeRange: TimeRange,
+    topN: string
+  ) => {
```

Comment:
I don't think there is any difference to debounce an async function or a normal function. They will all have the behavior that when it is called multiple times the call will be debounced. So how about simply:

```
async function queryXxx(...) {

}

const queryXxxDebounced = _.debounce(...)
```